### PR TITLE
Update to MyFaces 4.0.3

### LINKED ISSUE #30276: Fix MYFACES-4679 in 2.3 and Higher
**Describe the bug**  
Ajax Events Can Trigger Button Actions Unintentionally in 

**Steps to Reproduce**  
If some element is ajax-enabled and specifies an event, it may trigger the action of the encompassing element. 

For example, the blur event would trigger the bean.confirm action: 
```
<h:commandButton id="confirm2" value="listener" action="#{bean.confirm()}" >
 <f:ajax event="blur" listener="#{bean.listener()}"/> 
</h:commandButton>
```

**Expected behavior**  
The ajax enabled tag should not invoke any actions other than the listener, if specified. 

**Diagnostic information:**  
 - OpenLiberty Version: Since 24.0.0.1 
 - Affected feature(s): jsf-2.3, faces-3.0, faces-4.0 (jsf-2.2 was fixed via #29648)
 - Java Version: N/A
 - server.xml configuration (WITHOUT sensitive information like passwords)
 - If it would be useful, upload the messages.log file found in `$WLP_OUTPUT_DIR/messages.log`

**Additional context**

### LINKED ISSUE #31015: Update faces-4.0 to MyFaces 4.0.3
**Describe the use case that you want to enable:**
Please update faces-4.0 to the latest MyFaces 4.x release for Jakarta 10. Currently it is MyFaces 4.0.2.

**Describe why this is important to you:**
Identified a bug within PrimeFaces with https://github.com/primefaces/primefaces/issues/13578 which is bugged in MyFaces 4.0.1 and solved in 4.0.2.

### LINKED ISSUE #31094: Update to MyFaces 4.0.3
**Describe the bug**  
Update to the latest open source release to include the most recent bug fixes and improvements. 

Bug
    [[MYFACES-4605](https://issues.apache.org/jira/browse/MYFACES-4605)] - Cross form rending via ajax, form is missing ViewState
    [[MYFACES-4610](https://issues.apache.org/jira/browse/MYFACES-4610)] - Fix for issuing element key:value combination breaks Tobago
    [[MYFACES-4611](https://issues.apache.org/jira/browse/MYFACES-4611)] - NullPointerException in ViewPoolProcessor
    [[MYFACES-4623](https://issues.apache.org/jira/browse/MYFACES-4623)] - DuplicateIdException when adding component with resource in JSTL tag handler
    [[MYFACES-4624](https://issues.apache.org/jira/browse/MYFACES-4624)] - SearchExpressions must fail when component in viewroot is not resolvable
    [[MYFACES-4625](https://issues.apache.org/jira/browse/MYFACES-4625)] - PostRestoreStateEvent is not published
    [[MYFACES-4626](https://issues.apache.org/jira/browse/MYFACES-4626)] - Quarkus: Native compilation can't find key attribute
    [[MYFACES-4631](https://issues.apache.org/jira/browse/MYFACES-4631)] - DuplicateIdException via addComponentResource
    [[MYFACES-4633](https://issues.apache.org/jira/browse/MYFACES-4633)] - Suspicious Logic when setting BeanManager from JNDI
    [[MYFACES-4634](https://issues.apache.org/jira/browse/MYFACES-4634)] - Exception when processing attribute-extension property in faces-config.xml
    [[MYFACES-4636](https://issues.apache.org/jira/browse/MYFACES-4636)] - Using f:param + f:ajax onvent results in an error
    [[MYFACES-4638](https://issues.apache.org/jira/browse/MYFACES-4638)] - Inputs submitted by AJAX loose all content beginning with ampersand ("&") characters
    [[MYFACES-4640](https://issues.apache.org/jira/browse/MYFACES-4640)] - Smaller issues with the appendIussingElements code discovered by Tobago Tests
    [[MYFACES-4645](https://issues.apache.org/jira/browse/MYFACES-4645)] - Conversion error for select items of type String
    [[MYFACES-4646](https://issues.apache.org/jira/browse/MYFACES-4646)] - Incorrect formatting on String built by ValidatorException
    [[MYFACES-4647](https://issues.apache.org/jira/browse/MYFACES-4647)] - PushContextImpl send - NPE when no channel found for user
    [[MYFACES-4649](https://issues.apache.org/jira/browse/MYFACES-4649)] - ViewScoped bean handling broken because of refactoring
[[MYFACES-4427](https://issues.apache.org/jira/browse/MYFACES-4427)] - Cant open any pages in quarkus when using quarkus.package.type=uber-jar
    [[MYFACES-4642](https://issues.apache.org/jira/browse/MYFACES-4642)] - ViewScope doesn't work in non CDI environment
    [[MYFACES-4650](https://issues.apache.org/jira/browse/MYFACES-4650)] - UIViewRoot#resetValues called from #processPartialRendering should use VisitHint.SKIP_UNRENDERED
    [[MYFACES-4655](https://issues.apache.org/jira/browse/MYFACES-4655)] - Cannot resolver managed FacesValidator with empty value
    [[MYFACES-4656](https://issues.apache.org/jira/browse/MYFACES-4656)] - Component with overriden isRequired method does not evaluate plain string value
    [[MYFACES-4658](https://issues.apache.org/jira/browse/MYFACES-4658)] - faces.util.chain behavior changed
    [[MYFACES-4668](https://issues.apache.org/jira/browse/MYFACES-4668)] - Quarkus: Programmatic Views not working
    [[MYFACES-4671](https://issues.apache.org/jira/browse/MYFACES-4671)] - PushImpl.ts Socket#onerror attempting to use undefined variable
    [[MYFACES-4673](https://issues.apache.org/jira/browse/MYFACES-4673)] - Quarkus: Update to 3.8 LTS
    [[MYFACES-4675](https://issues.apache.org/jira/browse/MYFACES-4675)] - Faces #1936: resetValues() visit clientIds if not EditableValueHolder
    [[MYFACES-4676](https://issues.apache.org/jira/browse/MYFACES-4676)] - Error Data Paylod Not Following Spec in Faces 4.0
    [[MYFACES-4678](https://issues.apache.org/jira/browse/MYFACES-4678)] - Getting NPE FaceletStateValueExpression.getWrapped(jakarta.el.ELContext)
    [[MYFACES-4679](https://issues.apache.org/jira/browse/MYFACES-4679)] - MYFACES-4606 Causes Multiple Events To Queue
    [[MYFACES-4683](https://issues.apache.org/jira/browse/MYFACES-4683)] - Scripts executed in reverse order during ajax event
    [[MYFACES-4687](https://issues.apache.org/jira/browse/MYFACES-4687)] - Quarkus: Tomcat EL conflicting with Quakrus EL API
    [[MYFACES-4688](https://issues.apache.org/jira/browse/MYFACES-4688)] - Quarkus: Register more classes for reflection/serialization
    [[MYFACES-4694](https://issues.apache.org/jira/browse/MYFACES-4694)] - Quarkus: jakarta.faces.component._AttachedDeltaWrapper
    [[MYFACES-4695](https://issues.apache.org/jira/browse/MYFACES-4695)] - Ajax's render doesn't support nested elements
    [[MYFACES-4699](https://issues.apache.org/jira/browse/MYFACES-4699)] - faces.js response code string instead of number
    [[MYFACES-4703](https://issues.apache.org/jira/browse/MYFACES-4703)] - Infinite Redirect Loop if invalid Accept header
    [[MYFACES-4708](https://issues.apache.org/jira/browse/MYFACES-4708)] - Quarkus Compilcation Errors
    [[MYFACES-4709](https://issues.apache.org/jira/browse/MYFACES-4709)] - Faces Initialization for Apps Containing the FacesConfig Annotation
    [[MYFACES-4710](https://issues.apache.org/jira/browse/MYFACES-4710)] - CommandScript: Behaves differently than Mojarra
    [[MYFACES-4712](https://issues.apache.org/jira/browse/MYFACES-4712)] - ajax event listener looses 'source' in 'success' event
    [[MYFACES-4714](https://issues.apache.org/jira/browse/MYFACES-4714)] - Possible NPE in ClientWindowScopeContext#getContextualStorage
    [[MYFACES-4716](https://issues.apache.org/jira/browse/MYFACES-4716)] - Ajax is not working for empty selection for h:selectManyListbox

Improvements
    [[MYFACES-4604](https://issues.apache.org/jira/browse/MYFACES-4604)] - FacesInitializerImpl/AbstractFacesInitializer: Update f:websocket message
    [[MYFACES-4606](https://issues.apache.org/jira/browse/MYFACES-4606)] - Missing source button id:value pair from request parameters in ajax requests
    [[MYFACES-4614](https://issues.apache.org/jira/browse/MYFACES-4614)] - Print error message when action method not found
    [[MYFACES-4628](https://issues.apache.org/jira/browse/MYFACES-4628)] - Disabled Value Bug in 4.0 (Erroneous 'value is required' message)
    [[MYFACES-4630](https://issues.apache.org/jira/browse/MYFACES-4630)] - Compressed Files Necessary in 4.0 API?
[[MYFACES-4659](https://issues.apache.org/jira/browse/MYFACES-4659)] - Hidden field autocomplete="off" is not valid anymore
    [[MYFACES-4663](https://issues.apache.org/jira/browse/MYFACES-4663)] - Quarkus: Register RepeatStatus for reflection
    [[MYFACES-4691](https://issues.apache.org/jira/browse/MYFACES-4691)] - AJAX JS core should log a warning on the console, when component to update isnt found
    [[MYFACES-4700](https://issues.apache.org/jira/browse/MYFACES-4700)] - Javascript AJAX: displayError only when no faces.ajax.addOnError listener was registred
    [[MYFACES-4701](https://issues.apache.org/jira/browse/MYFACES-4701)] - Add ability to skip logging of specific exceptions in the ExceptionHandler impl

Task
    [[MYFACES-4648](https://issues.apache.org/jira/browse/MYFACES-4648)] - PushContextImpl - different behaviour when socket is not connected

**Steps to Reproduce**  
See individual JIRAs

**Expected behavior**  
See individual JIRAs

**Diagnostic information:**  
 - OpenLiberty Version: Since [23.0.0.7](https://github.com/OpenLiberty/open-liberty/releases/tag/gm-23.0.0.7)
 - Affected feature(s) faces-4.0
 - Java Version: N/A 
 - server.xml configuration N/A

**Additional context**  
Release Notes can be found below. This release includes fixes for both 4.0.2:
https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=10600&version=12353333
and 4.0.3:
https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=10600&version=12354227

### PR DESCRIPTION
**Pull request by volosied**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

(edit initially for  31094, 30276, and 31015 but reverted)

### CODE REVIEW

**Comment by volosied** [comment]
!build 
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=a7ca3bc1-5805-4e8d-afb3-864ca803f205) - :white_check_mark: completed successfully!)
fat.test.mode=full
fat.buckets.to.run=com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.2_fat.2,com.ibm.ws.jsf.2.3_fat,com.ibm.ws.jsf.2.3_fat.2,io.openliberty.org.apache.myfaces.4.0_fat

_Note: Target locations of links might be accessible only to IBM employees._

File: dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/tests/JSFSimpleHtmlUnit.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2015, 2024 IBM Corporation and others.
+ * Copyright (c) 2015, 2025 IBM Corporation and others.
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/tests/JSFSimpleHtmlUnit.java:2]: Why the update here?

Comment (pnicolucci) [reply dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/tests/JSFSimpleHtmlUnit.java:2]: Updated in a different PR

**Review by pnicolucci — APPROVED** [review]

**Comment by volosied** [comment]
#libby ([view Open Liberty Delivery Requirements Verification](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=2b07c436-9d3d-4fa3-90ce-82527014063c) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._
